pkg/storegateway/indexheader/index/postings.go

```diff
@@ -323,34 +323,42 @@ type postingValueOffsets struct {
 	lastValOffset int64
 }
 
-func (e *postingValueOffsets) prefixOffset(prefix string) (int, bool) {
+// prefixOffsets returns the index of the first matching offset (start) and the index of the first non-matching (end).
```

Comment:
[nit] I would clarify that "end" will be equal to "last index + 1" if all postings match the prefix.